lti_consumer/plugin/views.py

```diff
@@ -415,3 +418,130 @@ def scores(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsContextMembershipViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    LTI NRPS Context Membership Service endpoint.
+    """
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsContextMembershipsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    serializer_class = LtiNrpsContextMembershipSerializer
+
+    # Custom pagination class to support pagination via Link Header
+    pagination_class = LTINRPSMembersipPagination
+
+    def parse_role_filter(self):
+        """
+        Parse role filter from query param and map it to edx role.
+        """
+        role_filter_param = self.request.query_params.get('role', None)
+
+        matched_role = None
+        if role_filter_param:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if role_filter_param in context_roles:
+                    matched_role = edx_role
+
+        return matched_role
+
+    def get_memberships(self, kwarg_overrides=None):
+        """
+        Helper method to create queryset.
+        """
+        # get course key
+        course_key = self.request.lti_configuration.location.course_key
+
+        kwargs = {
+            'access_roles': [],
+            'include_students': True,
+        }
+
+        if kwarg_overrides and isinstance(kwarg_overrides, dict):
+            kwargs.update(kwarg_overrides)
+
+        return partial(compat.get_course_members, course_key, **kwargs)
+
+    def get_queryset(self):
+        """
+        Get User queryset related to current course. Also filter via role if provided.
+        """
+        kwargs = {}
+
+        # check if role filter has given
+        filter_by = self.parse_role_filter()
+
+        if filter_by is None:
+            # if no role filter given, we are interested in instructor & staff. other access
+            # roles like beta tester etc doesn't have equivalent role in LTI Context Role vocab.
+            kwargs['access_roles'] = ['instructor', 'staff']
+        elif filter_by != 'student':
+            kwargs['access_roles'] = [filter_by]
+            # exclude students if filtered by any access role
+            kwargs['include_students'] = False
+
+        return self.get_memberships(kwargs)
+
+    def preprocess(self, data):
+        """
+        Preprocess the output of get_membership method.
+        For example - append external ids for user.
+        """
```

Comment:
The only purpose of this is to add the external ids to the users, so rename this function accordingly.
```suggestion
    def attach_external_id_information(self, data):
        """
        A helper method that attaches the external id information to the user queryset.
        """
```

Also, add a better function docstring explaining the parameters and output.

Reply:
Renamed and updated docstring.